pvlib/snow.py

```diff
@@ -0,0 +1,182 @@
+
+    Notes
+    -----
+    Implements the model described in [1]_ with minor improvements in [2]_.
+
+    References
+    ----------
+    .. [1] Marion, B.; Schaefer, R.; Caine, H.; Sanchez, G. (2013).
+       "Measured and modeled photovoltaic system energy losses from snow for
+       Colorado and Wisconsin locations." Solar Energy 97; pp.112-121.
+    .. [2] Ryberg, D; Freeman, J. "Integration, Validation, and Application
+       of a PV Snow Coverage Model in SAM" (2017) NREL Technical Report
+    '''
+    timestep = _time_delta_in_hours(snowfall.index)
+    hourly_snow_rate = snowfall / timestep
+    # if we can infer a time frequency, use first snowfall value
+    # otherwise the first snowfall value is ignored
+    freq = pd.infer_freq(snowfall.index)
+    freq = freq.upper().replace('T', 'M')  # for pandas <= 0.23.4
+    if freq is not None:
+        if len(freq) == 1:  # only a unit abbreviation
+            freq = "1" + freq
+        timedelta = pd.Timedelta(freq).total_seconds() / 3600
+        hourly_snow_rate.iloc[0] = snowfall[0] / timedelta
+    else:  # can't infer frequency from index
+        hourly_snow_rate[0] = 0  # replaces NaN
+    return hourly_snow_rate > threshold_snowfall
+
+
+def coverage_nrel(snowfall, poa_irradiance, temp_air, surface_tilt,
+                  initial_coverage=0, threshold_snowfall=1.,
+                  can_slide_coefficient=-80., slide_amount_coefficient=0.197):
+    '''
+    Calculates the fraction of the slant height of a row of modules covered by
+    snow at every time step.
+
+    Implements the model described in [1]_ with minor improvements in [2]_,
+    with the change that the output is in fraction of the row's slant height
+    rather than in tenths of the row slant height. As described in [1]_, model
+    validation focused on fixed tilt systems.
+
+    Parameters
+    ----------
+    snowfall : Series
+        Accumulated snowfall within each time period. [cm]
+    poa_irradiance : Series
+        Total in-plane irradiance [W/m^2]
+    temp_air : Series
+        Ambient air temperature [C]
+    surface_tilt : numeric
+        Tilt of module's from horizontal, e.g. surface facing up = 0,
+        surface facing horizon = 90. Must be between 0 and 180. [degrees]
+    initial_coverage : float, default 0
+        Fraction of row's slant height that is covered with snow at the
+        beginning of the simulation. If None (default) then the initial
+        coverage is set to the snowfall in the first time period. [unitless]
+    threshold_snowfall : float, default 1.0
+        Hourly snowfall above which snow coverage is set to the row's slant
+        height. [cm/hr]
+    can_slide_coefficient : float, default -80.
+        Coefficient to determine if snow can slide given irradiance and air
+        temperature. [W/(m^2 C)]
+    slide_amount_coefficient : float, default 0.197
+        Coefficient to determine fraction of snow that slides off in each
+        time period. [unitless]
```

Comment:
This parameter is assumed to be hourly in the code, perhaps that should be mentioned here.